src/mds/Server.cc

```diff
@@ -3217,14 +3226,14 @@ void Server::apply_allocated_inos(MDRequestRef& mdr, Session *session)
   if (mdr->alloc_ino) {
     mds->inotable->apply_alloc_id(mdr->alloc_ino);
   }
-  if (mdr->prealloc_inos.size()) {
+  if (session && mdr->prealloc_inos.size()) {
     ceph_assert(session);
     session->pending_prealloc_inos.subtract(mdr->prealloc_inos);
     session->info.prealloc_inos.insert(mdr->prealloc_inos);
     mds->sessionmap.mark_dirty(session, !mdr->used_prealloc_ino);
     mds->inotable->apply_alloc_ids(mdr->prealloc_inos);
   }
-  if (mdr->used_prealloc_ino) {
+  if (session && mdr->used_prealloc_ino) {
     ceph_assert(session);
     session->info.used_inos.erase(mdr->used_prealloc_ino);
     mds->sessionmap.mark_dirty(session);
```

Comment:
I don't see code that tries taking inode number from inotable

Reply:
Allow_prealloc_inos is set to false as session is null.
Then ino=mds->inotable->project_alloc_id(). 
On line 3106